lib/langchain/llm/response/openai_response.rb

```diff
@@ -28,6 +28,13 @@ def tool_calls
       chat_completions&.dig(0, "message", "tool_calls")
     end
 
+    # @return [Hash] JSON schema structured response
+    def response_schema
+      if tool_calls
+        JSON.parse(tool_calls.first.dig("function", "arguments"))
```

Comment:
you have a module called `Langchain::LLM::OpenAIResponse` you should use it here.

Reply:
what happens when the LLM sends back an array of objects?

Reply:
It doesn't work with parallel function-calling right now if that's what you're asking.